library/Vanilla/Addon.php

```diff
@@ -531,13 +530,17 @@ private function scanDirPhp($dir) {
             yield $path;
         }
 
-        // Don't recursively scan the root of an addon.
-        if (empty($dir)) {
-            return;
-        }
+        $isDirRoot = $dir === '';
 
         // Get all of the php files from subdirectories.
         foreach ($this->glob("$dir/*", true) as $subdir) {
+            // Don't recursively scan the root of an addon unless it is uppercase.
+            // This means it may be part of a namespace.
+            $trimmedDir = ltrim($subdir, '/\\');
+            $isUppercaseDirName = strlen($trimmedDir) > 0 && ctype_upper($trimmedDir[0]);
+            if ($isDirRoot && !$isUppercaseDirName) {
+                continue;
+            }
```

Comment:
I'm okay approving your change if you don't want to do this, but it seems like the logic you are going with would be better put in `scanClassPaths()` to make it more complete. Like it can return all standard lowercase directories plus uppercase directories which would make it more complete and logical IMO.

Reply:
I agree actually. That would probably be a better solution. I'll get around to today